### PR TITLE
docs: rework CI a bit, more modern skipping

### PR DESCRIPTION
**Pull request by henryiii**
## Description

Minor docs modernization.

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -88,4 +88,6 @@ def build(session: nox.Session) -> None:
     session.log("Building normal files")
     session.run("python", "-m", "build", *session.posargs)
     session.log("Building pybind11-global files (PYBIND11_GLOBAL_SDIST=1)")
-    session.run("python", "-m", "build", *session.posargs, env={"PYBIND11_GLOBAL_SDIST": "1"})
+    session.run(
+        "python", "-m", "build", *session.posargs, env={"PYBIND11_GLOBAL_SDIST": "1"}
```
Comment (Skylion007) [inline noxfile.py:92]:
```suggestion
        sys.executable, "-m", "build", *session.posargs, env={"PYBIND11_GLOBAL_SDIST": "1"}
```

File: noxfile.py [hunk 2]
```diff
@@ -88,4 +88,6 @@ def build(session: nox.Session) -> None:
     session.log("Building normal files")
     session.run("python", "-m", "build", *session.posargs)
```
Comment (Skylion007) [inline noxfile.py:89]:
```suggestion
    session.run(sys.executable, "-m", "build", *session.posargs)
```

**Review by Skylion007 — CHANGES_REQUESTED** [review]

**Comment by Skylion007** [comment]
All the Python calls in the nox.py file should be replaced with sys.executable to ensure ti's calling the same Python module as Nox.

**Comment by henryiii** [comment]
You don't _want_ the same python calling nox inside the session; you want the one set up by the session inside the virtual environment nox creates. It explicitly always prepares the path, so `python` (or any other executable) is safe, and it will show a warning if anything external to the virtual environment is used (unless you pass `external=True`).

**Comment by Skylion007** [comment]
Ah, nvm then.

**Review by Skylion007 — APPROVED** [review]

**Pull request merged**